internal/provider/resource_azure_cloud_router_connection.go

```diff
@@ -215,7 +215,7 @@ func resourceAzureExpressRouteConnRead(ctx context.Context, d *schema.ResourceDa
 	_ = d.Set("azure_service_key", resp.CloudSettings.AzureServiceKey)
 	_ = d.Set("po_number", resp.PONumber)
 
-	if resp.CloudSettings.PublicIP != "" {
+	if resp.CloudSettings.PublicIP != "" || resp.CloudSettings.PrimaryPublicIP != "" || resp.CloudSettings.SecondaryPublicIP != "" {
 		_ = d.Set("is_public", true)
```

Comment:
I think we need to set the `primary_public_ip` and `secondary_public_ip` here so those are set in the TF state file.

```
_ = d.Set("primary_public_ip", resp.CloudSettings.PrimaryPublicIP)
_ = d.Set("primary_public_ip", resp.CloudSettings.SecondaryPublicIP)
```

Reply:
There's also a reference in the function flattenCloudSettings in datasource_cloud_connections.go which might need an update

Reply:
@vgvm-pf good call, you are correct!

Reply:
All done, thanks both!

Reply:
Cool and I think also in dataSourceCloudConnections schema in cloud_settings section. Starting to remember this stuff a little